### PR TITLE
Rendering: Set window dirty only on app thread

### PR DESCRIPTION
**Pull request by enen92**
## Description
Backport of https://github.com/xbmc/xbmc/pull/25121 (and https://github.com/xbmc/xbmc/pull/25132) for Omega

### CODE REVIEW

**Comment by ksooo** [comment]
Cannot say anything about correctness of the code, but I runtime-tested the PR and can confirm that it fixes the deadlock.

**Review by kambala-decapitator — APPROVED** [review]
looks good, but you may want to check this comment: https://github.com/xbmc/xbmc/pull/25121#discussion_r1590208699

File: xbmc/application/Application.cpp [hunk 1]
```diff
@@ -878,7 +878,16 @@ void CApplication::Render()
...
   // render gui layer
-  if (appPower->GetRenderGUI() && !m_skipGuiRender)
+  const bool renderGUI = appPower->GetRenderGUI();
+  if (m_guiRenderLastState != std::nullopt && renderGUI && m_guiRenderLastState != renderGUI)
+  {
+    CGUIComponent* gui = CServiceBroker::GetGUI();
+    if (gui)
+      CServiceBroker::GetGUI()->GetWindowManager().MarkDirty();
+  }
+  m_guiRenderLastState = renderGUI;
```
Comment (fritsch) [inline xbmc/application/Application.cpp:888]: I think you don't need this here. GetRenderGUI returns a simple bool, so the code can be simplified heavily. Might be I overlook some inheritance somewhere, but don't see it.

So the logic here is: whenever GetRenderGUI yields something that is not the current state it is marked dirty. Are you sure that this not simply disables dirtyregion entirely?

Normally I would assume - that we render when something is dirty. In this code though, we makr dirty whenever internal state and GetRenderGUI disagree.

Comment (enen92) [reply xbmc/application/Application.cpp]: Went with the suggestion from @sarbes for now so this doesn't apply anymore.

**Comment by ksooo** [comment]
@enen92 confirming 0f4a14d915a0cb19986f597b059787defed11dc1 working, runtime-tested on Omega.

**Pull request closed without merging**